### PR TITLE
add native staking delegate struct

### PR DESCRIPTION
**Pull request by dustinxie**
define the delegate struct for native staking
WIP on another PR for storing/accessing these delegates

### CODE REVIEW

File: action/protocol/staking/protocol.go [hunk 1]
```diff
@@ -39,8 +50,26 @@ func NewProtocol() *Protocol {
...
-	return nil, nil
+	// read native staking candidate list
+	if len(p.nativeCand) == 0 {
+		_, err := sm.State(
+			&p.nativeCand,
+			protocol.NamespaceOption(factory.StakingNameSpace),
+			protocol.KeyOption(factory.NativeCandKey))
+		if err != nil && errors.Cause(err) != state.ErrStateNotExist {
+			return nil, err
+		}
```
Comment (CoderZhi) [inline action/protocol/staking/protocol.go]: put in Start func

File: action/protocol/staking/protocol.go [hunk 2]
```diff
@@ -7,23 +7,34 @@
...
+var (
+	// ErrInvalidAmount indicates an invalid staking amount
+	ErrInvalidAmount = errors.New("invalid staking amount")
+)
+
 // Protocol defines the protocol of handling staking
 type Protocol struct {
-	addr address.Address
+	addr       address.Address
+	nativeCand state.CandidateList
```
Comment (CoderZhi) [inline action/protocol/staking/protocol.go]: convert to a map if possible

File: state/candidate.go [hunk 3]
```diff
@@ -29,13 +30,18 @@ var (
...
+type (
+	// CandName is the 12-byte candidate name
+	CandName [12]byte
+
+	// Candidate indicates the structure of a candidate
+	Candidate struct {
+		Address       string
+		Votes         *big.Int
+		RewardAddress string
+		CanName       []byte // used as identifier to merge with native staking result, not part of protobuf
```
Comment (CoderZhi) [inline state/candidate.go]: Should we restrict the type or it has been too late with backward compatible problem?

Comment (dustinxie) [reply state/candidate.go]: we could
but for safety (though the risk is low) would just leave it
we have func to12Bytes() to guarantee what we read from election must be 12-byte

File: action/protocol/staking/protocol.go (hunk 1 above)
Comment (dustinxie) [reply action/protocol/staking/protocol.go]: will fix in next PR

File: action/protocol/staking/protocol.go (hunk 2 above)
Comment (dustinxie) [reply action/protocol/staking/protocol.go]: fix in next PR

File: action/protocol/staking/bucket_index.go [hunk 4]
```diff
@@ -8,14 +8,13 @@ package staking
 
 import (
 	"github.com/golang/protobuf/proto"
-	"github.com/iotexproject/iotex-core/state"
-	"github.com/pkg/errors"
-
 	"github.com/iotexproject/go-pkgs/hash"
 	"github.com/iotexproject/iotex-address/address"
+	"github.com/pkg/errors"
```
Comment (lizhefeng) [inline action/protocol/staking/bucket_index.go]: Group imports

**Pull request merged**